interview-questions.md

```diff
@@ -8,42 +8,43 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a Float and an Integer in Ruby?
 
-Your answer:
+Your answer: A float is a number or integer that is a not a whole number, a good example is 3.14 or 4.1. An integer is a whole number such as 5 or 65. An integer and float are both considered data types. In order to return a float one of the numbers has to have a decimal. Floats and integers both have similar methods like JavaScript such as adding, subtracting, multiplying and dividing. 
+
+Researched answer: Integer and Float are both considered different data types. An integer or int is number without a decimal point. A float also known as floating-point number is a number that has a decimal place. Floats must be introduced as a float to your program such as 4.0/3 or 5.0 * 7.
 
-Researched answer:
 
 1. Ruby has an implicit return. What does that mean?
 
-Your answer:
+Your answer: In Ruby, an implicit return is basically returning the method without actually using the word "return". Unlike Ruby, JavaScript functions need a return for that value. For any method, the return in the Ruby language will be the last because of implicit retrurn. The logic is understood without having to use a word.
 
-Researched answer:
+Researched answer: The implicit return in Ruby is the final declaration inside that method, the method returns the final line of code before  "end" is displayed. In between def and end is the logic, implicit return will always return the last line of code. It is important to remember the last line before "end" will be returned for that method. 
 
 1. What is a block in Ruby?
 
-Your answer:
+Your answer: A block is the equivalent to a function in JavaScript. A block in the Ruby language starts with "do" and ends with "end." For blocks, curly braces can be used if the code is a one liner. Puts or p will invoke that block of code. A block of code is passed by top to bottom, meaning the last logic or data passed will be returned. 
 
-Researched answer:
+Researched answer: A block is a function that is passed inside a method that is interpreted with the keywords do and end. Built in methods such as map and filter will need more informatation therefore a block is needed to determine the outcome. Blocks are used to accept information from arguments and return a value.
 
 1. How do you extract a value in a Ruby hash?
 
-Your answer:
+Your answer: I will need to do some research on this question. 
 
-Researched answer:
+Researched answer: A hash is a collection of keys, hashes are designed like an array however the index in a hash would be the key. There are four actions taht can be performed on dtata in hashes.CRUD stands for create, update, read and delate data. By using the read action, users can extract either the whole value or individual values by using brackets. Various methods such as times,each,while and map require a block to ensure the desired outcome. 
```

Comment:
Awesome follow up! A hash can be thought of like a data structure very similar to an object in JS. I love that you mention keys because you are absolutely correct, that is how we extract the values instead of numbered indexes.